### PR TITLE
docs: Add type for icon property of guild

### PR DESCRIPTION
**Pull request by Blue-Robin-Taken**
## Summary
When I used this property, it took me a bit of time to find it's type. So, I believe it would be good to add it in the documentation.

## Information

- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [x] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [ ] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [ ] I have updated the changelog to include these changes.

### CODE REVIEW

File: discord/guild.py [hunk 1]
```diff
@@ -949,7 +949,7 @@ def owner(self) -> Member | None:
 
     @property
     def icon(self) -> Asset | None:
-        """Returns the guild's icon asset, if available."""
+        """Returns the guild's icon asset of type :class:`asset.Asset`, if available."""
```
Comment (Om1609) [inline discord/guild.py:952]: Can this not use the `rtype` directive? 
https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html#the-sphinx-docstring-format

**Review by Om1609 — CHANGES_REQUESTED** [review]

File: discord/guild.py (hunk 1 above)
Comment (mccoderpy) [reply discord/guild.py:952]: Something like :class:`~discord.Asset` should work fine. But I can't see any place in the PyCord docs where poperty return  types are mentioned in the docstring.

Comment (Lulalaby) [reply discord/guild.py:952]: @mccoderpy https://discord.com/channels/858089281214087179/1055198103369682966/1121542979769880637

Comment (Blue-Robin-Taken) [reply discord/guild.py:952]: Wouldn't it be :type:

Comment (mccoderpy) [reply discord/guild.py:952]: No, :type: isn't a thing by default, and types are mostly classes ^^

Comment (Om1609) [reply discord/guild.py:952]: `:type:` directive is to specify the type of a parameter. `:rtype` is to specify the return type.

See this https://www.sphinx-doc.org/en/master/tutorial/describing-code.html#documenting-python-objects

**Review by JustaSqu1d — COMMENTED** [review]
This doesn't seem to tackle the main problem. Currently, all properties do not have types, and this doesn't actually solve the core issue.

**Comment by JustaSqu1d** [comment]
> This doesn't seem to tackle the main problem. Currently, all properties do not have types, and this doesn't actually solve the core issue.

See #2141

**Pull request closed without merging**